docs/security/index.md

```diff
@@ -0,0 +1,400 @@
+the `SecurityPlugin.prepareLabelsApplicator(byte[], Graph)` method providing the labels byte sequence from the
+`Security-Label` header and the fine-grained labels graph (if any).  Therefore plugins **MUST** provide a fresh instance
+of this interface every time their `prepareLabelsApplicator()` method is invoked as the labels that apply **MAY** change
+on every RDF event an application processes.
+
+Once an application has an instance they call `labelForTriple(Triple)` whenever they need to lookup the label for a
+particular triple, this returns a `SecurityLabels<?>` from which the application can then store the corresponding byte
+sequence by accessing its `encoded()` method.
+
+### `Authorizer`
+
+The `Authorizer` allows an application to make access decisions based on the labels for data.  This is intended to be
+scoped to the lifetime of a single user request so an instance is created by calling the
+`SecurityPlugin.prepareAuthorizer(Entitlements<?>)` method passing in the users entitlements (see [Example
+Usage](#example-usage)) for a fuller outline usage.
+
+## Example Usage
+
+In the following example we show the complete lifecycle of how an application would make an access decision:
+
+```java
+// Load the plugin, this is cached for the lifetime of the JVM so repeated calls are cheap
+SecurityPlugin plugin = SecurityPluginLoader.load();
+
+// Get the Users Entitlements and prepare an Authorizer based upon those
+String userId = plugin.identityProvider().identityForUser(jws);
+Entitlements<?> entitlements = plugin.entitlementsProvider().entitlementsForUser(userId);
+Authorizer authorizer = plugin.prepareAuthorizer(entitlements);
+
+// Get the Labels Parser
+SecurityLabelsParser parser = plugin.labelsParser();
+
+// Get results from the underlying data store
+List<SomeResult> results = runQuery();
+for (int i = 0; i < results.size(); i++) {
+  // Filter the results based on the labels present
+  byte[] rawLabels = results.get(i).getSecurityLabels();
+  try {
+    SecurityLabels<?> labels = parser.parseSecurityLabels(rawLabels);
+    if (!authorizer.canAccess(labels)) {
+        results.removeAt(i);
+        i--;
+    }
+  } catch (MalformedLabelsException e) {
+    // Could be thrown if the labels are not supported by this plugin, or otherwise malformed
+    results.removeAt(i);
+    i--;
+  }
+}
+
+// Return the filtered results
+return results;
+```
+
+Firstly an application has to obtain the plugin instance, most applications will probably do this once early in their
+startup and make the instance accessible wherever it is needed.  It then uses other parts of the plugin API to determine
+the user identity, obtain their entitlements and prepare an authorizer based upon those.  Once it has an authorizer it
+can then make any access decisions necessary to fulfill the current request/operation before producing the filtered
+results.
```

Comment:
- [ ] Add clarifying note about `Authorizer` lifecycle here and recommend `try-with-resources` pattern